### PR TITLE
Token Claim Page

### PR DESCRIPTION
**Pull request by PonyJackal**
- added token claim page
- implemented token claim contract

### CODE REVIEW

File: src/client/js/utils/tokenList.js [hunk 1]
```diff
@@ -119,22 +119,24 @@ window.TokenListManager = {
...
     if (optionalNetwork) {
       tokenList = this.getTokenListForNetwork(optionalNetwork);
     }
 
     const foundToken = _.find(tokenList, function (v) {
       return (
-        v.address.toLowerCase() === tid.toLowerCase() ||
-        v.symbol.toLowerCase() === tid.toLowerCase()
+        v.address.toLowerCase() === tokenId.toLowerCase() ||
+        v.symbol.toLowerCase() === tokenId.toLowerCase()
```
Comment (chillvybz) [inline src/client/js/utils/tokenList.js:132]: you can revert these changes in this file. it was a bug in the main branch, which is now fixed.
that's why you did this correct?

File: src/client/js/utils/wallet.js [hunk 2]
```diff
@@ -350,6 +350,7 @@ window.WalletJS = {
   },
 
   _connectProviderMetamask: function () {
+    console.log("connect metamask")
```
Comment (chillvybz) [inline src/client/js/utils/wallet.js:353]: please remove

File: src/client/js/utils/tokenList.js (hunk 1 above)
Comment (PonyJackal) [reply src/client/js/utils/tokenList.js:132]: Yes, got your point

File: src/client/js/components/pages/TokenClaimHome.jsx [hunk 3]
```diff
@@ -1,27 +1,245 @@
...
+      );
+    }
+    return <TokenClaimDisconnectedWallet onClick={handleConnect} />;
+  };
+
+  return (
+    <div className="container">
+      <Navbar />
+      <MobileMenu />
+      <NotificationSystem />
```
Comment (chillvybz) [inline src/client/js/components/pages/TokenClaimHome.jsx:234]: let's remove NotificationSystem  and TxHistoryModal  in the stack here. not needed

File: src/client/js/components/pages/TokenClaimHome.jsx [hunk 4]
```diff
@@ -1,27 +1,245 @@
...
+                          <button
+                            className="button is-success token-claim-btn"
+                            onClick={claimTokens}
+                          >
+                            Claim Tokens
+                          </button>
+                        </div>
+                        <div className='token-claim-detail-contact'>
+                          <p>Having issues claiming your Swing Tokens?</p>
+                          <p className='token-claim-contact-us'>Contact us on Telegram</p>
```
Comment (chillvybz) [inline src/client/js/components/pages/TokenClaimHome.jsx:185]: use this Discord link instead: https://discord.gg/jQ9Xhdbb

File: src/client/js/components/partials/TokenClaimResultModal.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+                  <button onClick={props.handleClose} className="button is-success">
+                    Done
+                  </button>
+                </div>
+              : <div className="success-modal-body">
+                  <img src="/images/failture_icon.svg" alt="Something went wrong" />
+                  <p className="success-text">Something went wrong</p>
+                  <p>Having issues claiming your Swing Tokens?</p>
+                  <p className="token-claim-contact-us">Contact us on Telegram</p>
```
Comment (chillvybz) [inline src/client/js/components/partials/TokenClaimResultModal.jsx:32]: https://discord.gg/jQ9Xhdbb discord link instead

File: src/client/js/components/partials/TokenClaimResultModal.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        <div className="modal-content">
+          <div
+            className="box modal-basic-style modal-dropdown-options"
+          >
+            <div className="modal-body">
+              {props.success 
+              ? <div className="success-modal-body">
+                  <img src="/images/success_icon.svg" alt="Success" />
+                  <p className="success-text">Success</p>
+                  <p>Your $Swing tokens have been <br /> successfully added to your wallet.</p>
```
Comment (chillvybz) [inline src/client/js/components/partials/TokenClaimResultModal.jsx:22]: capitalize $SWING

**Review by chillvybz — CHANGES_REQUESTED** [review]
Almost there

**Comment by chillvybz** [comment]
PLease also resolve conflicts. You just need to revert `src/client/js/utils/tokenList.js` changes in this branch

**Comment by PonyJackal** [comment]
> PLease also resolve conflicts. You just need to revert `src/client/js/utils/tokenList.js` changes in this branch

Done

File: src/client/js/components/pages/TokenClaimHome.jsx [hunk 7]
```diff
@@ -1,27 +1,239 @@
...
+  }
+
+  const closeErrorModal = () => {
+    setHasError(false);
+  }
+
+  const renderTokenClaimHome = () => {
+    if (Wallet.isConnectedToAnyNetwork()) {
+      if (!balances.length && currentNetwork === undefined) {
+        return <EmptyBalances />;
```
Comment (chillvybz) [inline src/client/js/components/pages/TokenClaimHome.jsx:135]: Actually why is this here? Why need to show balances?

File: src/client/js/components/pages/TokenClaimHome.jsx [hunk 8]
```diff
@@ -1,27 +1,239 @@
...
+
+  const handleWalletChange = () => {
+    setMyApplicationState((prevState) => ({
+      ...prevState,
+      refresh: Date.now(),
+      balances: [],
+      loading: true,
+    }));
+
+    loadBalances();
```
Comment (chillvybz) [inline src/client/js/components/pages/TokenClaimHome.jsx:72]: Actually why is this here? Why need to show balances?

File: src/client/js/components/pages/TokenClaimHome.jsx [hunk 9]
```diff
@@ -1,27 +1,239 @@
...
+  useEffect(() => {
+    setMyApplicationState((prevState) => ({
+      ...prevState,
+      currentNetwork: TokenListManager.getCurrentNetworkConfig(),
+      refresh: Date.now(),
+      balances: [],
+      loading: true,
+    }));
+
+    loadBalances();
```
Comment (chillvybz) [inline src/client/js/components/pages/TokenClaimHome.jsx:51]: Actually why is this here? Why need to show balances?

Comment (chillvybz) [reply src/client/js/components/pages/TokenClaimHome.jsx:51]: Please remove balances everywhere in this file, if not needed

**Comment by chillvybz** [comment]
Also merge `development` one more time. I fixed the broken tests on the main branch.

**Comment by PonyJackal** [comment]
> Also merge `development` one more time. I fixed the broken tests on the main branch.

Done
